modules/services/borgmatic.nix

```diff
@@ -0,0 +1,298 @@
+        default = { };
+      };
+    };
+  };
+
+  # Move attrSet.extraConfig attributes to the top-level of attrSet:
+  inlineExtraConfig = attrSet:
+    (removeAttrs attrSet [ "extraConfig" ]) // attrSet.extraConfig;
+
+  removeNullValues = attrSet: filterAttrs (key: value: value != null) attrSet;
+
+  cleanupConfig = config: removeNullValues (inlineExtraConfig config);
+
+  writeConfig = config:
+    generators.toYAML { } {
+      location = cleanupConfig config.location;
+      storage = cleanupConfig config.storage;
+      retention = cleanupConfig config.retention;
+      consistency = cleanupConfig config.consistency;
+    };
+in {
+  meta.maintainers = [ maintainers.DamienCassou ];
+
+  options = {
+    services.borgmatic = {
+      enable = mkEnableOption "Borgmatic service";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.borgmatic;
+        defaultText = "pkgs.borgmatic";
+        description = ''
+          The borgmatic package to run.
+        '';
+      };
+
+      frequency = mkOption {
+        type = types.str;
+        default = "hourly";
+        description = ''
+          How often to run borgmatic.  This value is passed to the systemd
+          timer configuration as the onCalendar option.  See
+          <citerefentry>
+            <refentrytitle>systemd.time</refentrytitle>
+            <manvolnum>7</manvolnum>
+          </citerefentry>
+          for more information about the format.
+        '';
+      };
+
+      configs = mkOption {
+        type = types.attrsOf configModule;
+        description = ''
+          Borgmatic allows for several named backup configurations,
+          each with its own source directories and repositories.
+        '';
+        example = literalExpression ''
+          {
+            personal = {
+              location = {
+                source_directories = [ "/home/me/personal" ];
+                repositories = [ "ssh://myuser@myserver.com/./personal-repo" ];
+              };
+            };
+            work = {
+              location = {
+                source_directories = [ "/home/me/work" ];
+                repositories = [ "ssh://myuser@myserver.com/./work-repo" ];
+              };
+            };
+          };
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.borgmatic" pkgs
+        lib.platforms.linux)
+    ];
+
+    # ."borgmatic.d/first.yaml".text = writeConfig cfg.config;
+    xdg.configFile = with lib.attrsets;
+      mapAttrs' (configName: config:
+        nameValuePair ("borgmatic.d/" + configName + ".yaml") {
+          text = writeConfig config;
+        }) cfg.configs;
+
+    home.packages = [ cfg.package ];
+
+    systemd.user = {
+      services.borgmatic = {
+        Unit = {
+          Description = "borgmatic backup";
+          Wants = [ "network-online.target" ];
+          After = [ "network-online.target" ];
```

Comment:
Unfortunately, `network-online.target` is not available for the user systemd session so I think this will start with no consideration made to network status.

Reply:
I didn't know that, thank you. Removed.